contracts/SynthetixRouterStrategy.sol

```diff
@@ -0,0 +1,351 @@
+                exchangeSynthToSUSD(newBalanceOfSynth);
+            }
+        }
+
+        uint256 totalAssets = balanceOfWant();
+        if (_amount > totalAssets) {
+            _liquidatedAmount = totalAssets;
+            _loss = _amount.sub(totalAssets);
+        } else {
+            _liquidatedAmount = _amount;
+        }
+    }
+
+    function liquidateAllPositions()
+        internal
+        override
+        returns (uint256 _amountFreed)
+    {
+        // In order to work, manualRemoveFullLiquidity needs to be call 6 min in advance
+        require(checkWaitingPeriod(), "Wait for settlement period");
+        require(
+            valueOfInvestment() == 0,
+            "Need to remove liquidity from vault first"
+        );
+        require(_balanceOfSynth() == 0, "Need to exchange synth to want first");
+        _amountFreed = balanceOfWant();
+    }
+
+    function manualRemoveFullLiquidity()
+        external
+        onlyVaultManagers
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        // It will remove max amount of assets and trade synth for sUSD
+        // the Synthetix waiting period will start (and harvest can be called 6 mins later)
+        if (yVault.balanceOf(address(this)) > 0) {
+            super.liquidateAllPositions();
+        }
+        // It will withdraw all the assets from the yvault and the exchange them to want
+        (_liquidatedAmount, _loss) = withdrawSomeWant(
+            estimatedTotalAssets(),
+            true
+        );
+    }
+
+    function manualRemoveLiquidity(uint256 _liquidityToRemove)
+        external
+        onlyVaultManagers
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        _liquidityToRemove = Math.min(
+            _liquidityToRemove,
+            estimatedTotalAssets()
+        );
+        // It will withdraw all the assets from the yvault and the exchange them to want
+        (_liquidatedAmount, _loss) = withdrawSomeWant(_liquidityToRemove, true);
+    }
+
+    function estimatedTotalAssets() public view override returns (uint256) {
+        return
+            _balanceOfSUSD().add(_synthToSUSD(_balanceOfSynth())).add(
+                valueOfInvestment()
+            );
+    }
+
+    function _ethToWant(uint256 _amount) internal view returns (uint256) {
+        address[] memory path = new address[](2);
+        path[0] = weth;
+        path[1] = address(want);
+
+        uint256[] memory amounts =
+            IUni(uniswapRouter).getAmountsOut(_amount, path);
+
+        return amounts[amounts.length - 1];
+    }
+
+    function valueOfInvestment() public view override returns (uint256) {
+        return
+            _synthToSUSD(
+                yVault.balanceOf(address(this)).mul(yVault.pricePerShare()).div(
+                    10**yVault.decimals()
+                )
+            );
+    }
+
+    function prepareMigration(address _newStrategy) internal override {
+        super.prepareMigration(_newStrategy);
```

Comment:
does it make sense to use this instead of just withdrawing from the vault? 

it kind of makes it less readable

Reply:
Yes, similar to the next comment. I prefer using classes and inheritance to address this, but if we value more readability I can change it.